File: mobile/lib/features/home/domain/product.dart
class Product {
  final String id;
  final String name;
  final String description;
  final double basePrice;
  final double discountPrice;
  final String categoryName;
  final int stock;
  final double rating;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.basePrice,
    required this.discountPrice,
    required this.categoryName,
    required this.stock,
    required this.rating,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'] ?? '',
      basePrice: (json['basePrice'] as num).toDouble(),
      discountPrice: (json['discountPrice'] as num).toDouble(),
      categoryName: json['categoryName'] ?? '',
      stock: json['stock'] ?? 0,
      rating: (json['rating'] as num?)?.toDouble() ?? 0.0,
    );
  }
}

File: backend/src/AmazonClone.Domain/Entities/Product.cs
using System;
using System.Collections.Generic;
using AmazonClone.Domain.Common;

namespace AmazonClone.Domain.Entities
{
    public class Product : BaseEntity
    {
        public Guid SellerId { get; set; }
        public Seller? Seller { get; set; }

        public Guid CategoryId { get; set; }
        public Category? Category { get; set; }

        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal BasePrice { get; set; }
        public decimal DiscountPrice { get; set; }
        public int Stock { get; set; }
        
        public bool IsApproved { get; set; } = false;

        public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
        public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}
